format/FormatNexusJungfrauExt.py

```diff
@@ -0,0 +1,123 @@
+import h5py
+import numpy as np
+
+from xfel.util.jungfrau import pad_stacked_format
+
+from dials.array_family import flex
+
+from dxtbx.format.FormatNexus import FormatNexus
+
+"""
+This class can read a Jungfrau image file that includes, in addition to the corrected data,
+the gain maps, and the dark (pedestal) averages and RMS, providing the user access
+to that data.
+This was written to process SWISSFEL image files created using the CCTBX script jf16m_cxigeom2nexus.py
+
+  libtbx.python cctbx_project/xfel/swissfel/jf16m_cxigeom2nexus.py \
+  unassembled_file=X \
+  raw_file=Y \
+  geom_file=Z \
+  pedestal_file=P \
+  gain_file=G \
+  detector_distance=D \
+  include_spectra=True \
+  output_file=O \
+  raw=False \
+  beam_file=B
+
+where X,Y are the corrected,raw hdf5 files obtained at SWISSFEL for a particular run, Z is the crystFEL geom file
+provided by SWISSFEL for the jungfrau, P is the hdf5 file provided by swissfel containing the pedestal data,  G is
+the gainmap hdf5 file provided by SWISSFEL, D is the detector distance in mm, O is an output filename (to be read by
+this format class) and B is the hdf5 file provided by SWISSFEL containing the wavelength spectra measurements (optional)
+"""
+
+
+class FormatNexusJungfrauExt(FormatNexus):
+    @staticmethod
+    def understand(image_file):
```

Comment:
Being defensive here, is there not a Jungfrau detector name in the NeXus file?

Reply:
[Yah there is](https://github.com/cctbx/cctbx_project/blob/master/xfel/swissfel/jf16m_cxigeom2nexus.py#L69).  We can check for it.

Reply:
Are you suggesting we re-name the format class ? Or rather only understand the format if the name is "SwissFEL ARAMIS BEAMLINE ESB" ?

Reply:
The format class name I think is fine, an extended implementation of the Jungfrau data.  But for now we only think it's useful for the SwissFEL data, so we can check the name in the understand method.  If we determine the class is more generally useful, we can always back this out.

Reply:
Done!